dojo/group/utils.py

```diff
@@ -51,6 +80,20 @@ def group_post_delete_handler(sender, **kwargs):
         group.auth_group.delete()
 
 
+@receiver(post_delete, sender=Group)
+def auth_group_post_delete_handler(sender, **kwargs):
+    group = kwargs.pop('instance')
+    # DefectDojo group doesn't get deleted automatically when underlying group
+    # is deleted. we need to avoid post_delete recursion.
+    try:
+        dgroup = Dojo_Group.objects.get(name=group.name)
+        dgroup.auth_group = None
+        dgroup.save()
+        dgroup.delete()
```

Comment:
Unfortunately, this will cause recursion, see the comments and code above this change.
I'll try to clarify additionally. When you delete `Dojo_Group` (`dgroup.delete()`), `group_post_delete_handler` is called. This handler attempts to delete `dgroup.auth_group` (`group.auth_group.delete()` on line 80). This triggers `auth_group_post_delete_handler` which again tries to fetch `Dojo_Group` (not deleted yet) and the process continues.

In order to break this loop and avoid such unclear `post_something` implementation some major refactoring is needed, from my point of view, which would handle authentication-related objects on completely different level. Some inspiration: https://github.com/netbox-community/netbox/issues/5442 . Personally, I am not skilled enough to implement such change.